### PR TITLE
fix: don't handle invalid source map

### LINKED ISSUE #261: Couldn't parse bundle asset "vendors~typeahead.chunk.min.js". Analyzer will use module sizes from stats file.
Plugin version 1.2.4
webpack 4

This only happens in production when setting ` sourceMap: false`, when set to true, it runs fine!

```
 new UglifyJsPlugin({ 
            sourceMap: false, // <-- using false here, generates the error above
            uglifyOptions: {
                ecma: 5,
                warnings: true
            }
        })
```

**Here the complete Log-file:**
[2018-03-16T16_31_25_817Z-debug.log](https://github.com/webpack-contrib/uglifyjs-webpack-plugin/files/1820016/2018-03-16T16_31_25_817Z-debug.log)

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes #261

Also refactor tests

### CODE REVIEW

**Comment by d3viant0ne** [comment]
Please rebase with current master

**Comment by evilebottnawi** [comment]
@d3viant0ne done, something wrong with ci on node6 :confused:

File: src/index.js [hunk 1]
```diff
@@ -122,9 +123,18 @@ class UglifyJsPlugin {
...
 
-              sourceMap = new SourceMapConsumer(inputSourceMap);
+              if (utils.isSourceMap(map)) {
+                inputSourceMap = map;
+                sourceMap = new SourceMapConsumer(inputSourceMap);
+              } else {
+                inputSourceMap = map;
+                sourceMap = null;
+
+                compilation.warnings.push(
```
Comment (evilebottnawi) [inline src/index.js]: Throw warning on invalid source map. Is it very DX.

File: test/parallel-options.test.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ describe('when options.parallel', () => {
         beforeEach(() => {
           chunkPluginEnvironment = new PluginEnvironment();
           compilation = chunkPluginEnvironment.getEnvironmentStub();
-          compilation.assets = assets;
+          compilation.assets = Object.assign({}, assets);
```
Comment (evilebottnawi) [inline test/parallel-options.test.js]: Compilation mondify assets, we should clone assets

**Pull request merged**